diff --git a/node_modules/@react-native-community/audio-toolkit/ios/ReactNativeAudioToolkit/ReactNativeAudioToolkit/AudioRecorder.m b/node_modules/@react-native-community/audio-toolkit/ios/ReactNativeAudioToolkit/ReactNativeAudioToolkit/AudioRecorder.m
index 80d71e8..fd78bed 100644
--- a/node_modules/@react-native-community/audio-toolkit/ios/ReactNativeAudioToolkit/ReactNativeAudioToolkit/AudioRecorder.m
+++ b/node_modules/@react-native-community/audio-toolkit/ios/ReactNativeAudioToolkit/ReactNativeAudioToolkit/AudioRecorder.m
@@ -25,6 +25,12 @@ @implementation AudioRecorder
 
 @synthesize bridge = _bridge;
 
+id _powerUpdateTimer;
+int _frameId;
+NSNumber *_currentRecorderId;
+NSDate *_prevPowerUpdateTime;
+int _powerUpdateInterval = 250;
+
 - (void)dealloc {
     AVAudioSession *audioSession = [AVAudioSession sharedInstance];
     NSError *error = nil;
@@ -127,6 +133,8 @@ -(NSNumber *) keyForRecorder:(nonnull AVAudioRecorder*)recorder {
         return;
     }
     
+    recorder.meteringEnabled = YES;
+    
     callback(@[[NSNull null], filePath]);
 }
 
@@ -143,6 +151,10 @@ -(NSNumber *) keyForRecorder:(nonnull AVAudioRecorder*)recorder {
         callback(@[dict]);
         return;
     }
+    
+    _currentRecorderId = recorderId;
+    [self startMonitorPower];
+
     callback(@[[NSNull null]]);
 }
 
@@ -210,4 +222,43 @@ - (void)audioRecorderEncodeErrorDidOccur:(AVAudioRecorder *)recorder
                                                       }];
 }
 
+- (void)startMonitorPower {
+    _prevPowerUpdateTime = nil;
+    [self stopMonitorPower];
+
+    _powerUpdateTimer = [CADisplayLink displayLinkWithTarget:self selector:@selector(sendPowerUpdate)];
+    [_powerUpdateTimer addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSDefaultRunLoopMode];
+}
+
+- (void)stopMonitorPower {
+    [_powerUpdateTimer invalidate];
+}
+
+- (void)sendPowerUpdate {
+    AVAudioRecorder *recorder = [[self recorderPool] objectForKey:_currentRecorderId];
+    if (!recorder || !recorder.isRecording) {
+        return;
+    }
+
+    if (_prevPowerUpdateTime == nil ||
+        (([_prevPowerUpdateTime timeIntervalSinceNow] * -1000.0) >= _powerUpdateInterval)) {
+        _frameId++;
+        NSMutableDictionary *data = [[NSMutableDictionary alloc] init];
+        [data setObject:[NSNumber numberWithFloat:_frameId] forKey:@"id"];
+
+        [recorder updateMeters];
+        float _currentLevel = [recorder averagePowerForChannel: 0];
+        [data setObject:[NSNumber numberWithFloat:_currentLevel] forKey:@"value"];
+        [data setObject:[NSNumber numberWithFloat:_currentLevel] forKey:@"rawValue"];
+
+        NSString *eventName = [NSString stringWithFormat:@"RCTAudioRecorderEvent:%@", _currentRecorderId];
+        [self.bridge.eventDispatcher sendAppEventWithName:eventName
+                                                   body:@{@"event": @"power",
+                                                          @"data" : data
+                                                          }];
+
+        _prevPowerUpdateTime = [NSDate date];
+    }
+}
+
 @end
diff --git a/node_modules/@react-native-community/audio-toolkit/src/Recorder.js b/node_modules/@react-native-community/audio-toolkit/src/Recorder.js
index 69e679c..7cfda30 100644
--- a/node_modules/@react-native-community/audio-toolkit/src/Recorder.js
+++ b/node_modules/@react-native-community/audio-toolkit/src/Recorder.js
@@ -27,11 +27,12 @@ var defaultRecorderOptions = {
  * @constructor
  */
 class Recorder extends EventEmitter {
-  constructor(path, options = defaultRecorderOptions) {
+  constructor(path, options = defaultRecorderOptions, onNewPower) {
     super();
 
     this._path = path;
     this._options = options;
+    this._onNewPower = onNewPower;
 
     this._recorderId = recorderId++;
     this._reset();
@@ -67,6 +68,11 @@ class Recorder extends EventEmitter {
         this._reset();
         //this.emit('error', data);
         break;
+      case 'power':
+        if (this._onNewPower) {
+          this._onNewPower(data);
+        }
+        break;
     }
 
     this.emit(event, data);
